ui/src/app/app.tsx

```diff
@@ -144,6 +157,20 @@ export class App extends React.Component<{}, {popupProps: PopupProps; showVersio
             link.type = 'text/css';
             document.head.appendChild(link);
         }
+
+        const systemExtensions = services.extensions.getSystemExtensions();
+        const extendedNavItems = this.navItems;
+        const extendedRoutes = this.routes;
+        for (const extension of systemExtensions) {
+            extendedNavItems.push({
+                title: extension.title,
+                path: extension.path,
+                iconClassName: `fa ${extension.icon}`
+            });
+            extendedRoutes[extension.path] = {component: extension.component as React.ComponentType<React.ComponentProps<any>>, extension: true};
```

Comment:
Noticed one minor but annoying issue. When the user navigates to extension page browser keep showing previous page title. 
I think we can fix it by wrapping `extension.component` into `Helmet` component:

```typescript
component: (<>
  <Helmet>
     <title>{extension.title} - {Argo CD}</title>
  </Helmet>
  {extension.component}
</>)
```